src/Simplify/AstHelpers.elm

```diff
@@ -952,32 +952,36 @@ isSpecificUnappliedBinaryOperation symbol checkInfo expression =
             False
 
 
-{-| Indicates whether this value is potentially NaN in the context of a Dict key or Set value, meaning that it could return `False` when `==` with itself.
+{-| Indicates whether this value is potentially NaN,
+meaning that it could return `False` when `==` with itself.
 
-This will return `False` for expressions that are known to not be `comparable` (e.g. records) nor `numbers`.
+This will return `False` for expressions that are known to
+only contain literals (e.g. [ ( 0, { name = "string" } ) ]),
+are known operations that never produce NaN (e.g. `a ++ b`),
```

Comment:
`a ++ b` can produce NaN because you could have `[ NaN ] ++ []` which is itself comparable.